CDCgov/prime-reportstream
# Fixed bug to allow custom FHIR path functions in filters

This PR fixes a bug to allow custom FHIR path functions in filters. As part of the CA Flu pilot we need to route Flu only results to CA. The conditionFilter for CA was modified to exclude only Covid results, instead of having to list all the LOINC CA wants. Maintaining that list was going to start getting too complicated with the addition of Flu only results, RSV, and HIV/STI.

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Test Steps:
1. Check out and build the branch
2. run `./gradlew reloadSettings`
3. Send a test report using the attached Postman collection
[Simple Report Bug Fix.postman_collection.txt](https://github.com/CDCgov/prime-reportstream/files/12196799/Simple.Report.Bug.Fix.postman_collection.txt)

4. Verify that the request was accepted
5. Open the generated HL7 file in the local sftp folder.
6. Verify that only Flu results are routed to CA.

## Changes
- Fixed bug to allow custom FHIR path function in filters
- Updated CA conditionFilter to exclude Covid results

## Checklist

### Testing
- [] Added tests?

### Process

## Linked Issues
- Addresses #10635
